xDrip/Services/CloudSync/Codables/CTreatment.swift

```diff
@@ -0,0 +1,258 @@
+            modifyRequestType = .modifyBasal
+            deleteRequestType = .deleteBasal
+        case .training:
+            postrequestType = .postTraining
+            modifyRequestType = .modifyTraining
+            deleteRequestType = .deleteTraining
+        }
+        
+        switch requestType {
+        case .post:
+            return postrequestType
+        case .modify:
+            return modifyRequestType
+        case .delete:
+            return deleteRequestType
+        }
+    }
+}
+
+struct CTreatment: Codable {
+    var type: TreatmentType?
+    let treatmentID: String?
+    let carbs: Double?
+    let duration: Double?
+    let exerciseIntensity: String?
+    let eventType: String?
+    let insulin: Double?
+    let uuid: String?
+    let createdAt: String?
+    let sysTime: String?
+    let utcOffset: Int?
+    let timestamp: Int?
+    let enteredBy: String?
+    let insulinInjections: String?
+    let foodType: String?
+    
+    enum CodingKeys: String, CodingKey {
+        case treatmentID = "_id"
+        case carbs
+        case eventType
+        case insulin
+        case uuid
+        case createdAt = "created_at"
+        case sysTime
+        case utcOffset
+        case timestamp
+        case enteredBy
+        case insulinInjections
+        case foodType
+        case duration
+        case exerciseIntensity
+    }
+    
+    init(entry: TreatmentEntryProtocol, treatmentType: TreatmentType) {
+        type = treatmentType
+        
+        uuid = entry.externalID
+        treatmentID = CTreatment.getIDFromUUID(uuid: uuid)
+        
+        switch treatmentType {
+        case .carbs:
+            carbs = entry.amount
+            insulin = nil
+            duration = nil
+            exerciseIntensity = nil
+        case .bolus, .basal:
+            carbs = nil
+            insulin = entry.amount
+            duration = nil
+            exerciseIntensity = nil
+        case .training:
+            carbs = nil
+            insulin = nil
+            duration = entry.amount / .secondsPerMinute
+            
+            if let intensityValue = entry.exerciseIntensity,
+               let intensityString = TrainingIntensity(rawValue: intensityValue)?.paramValue() {
+                exerciseIntensity = intensityString
+            } else {
+                exerciseIntensity = nil
+            }
+        }
+        
+        eventType = treatmentType.rawValue.capitalized
+        
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        sysTime = dateFormatter.string(from: entry.date ?? Date())
+        
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss'Z'"
+        dateFormatter.timeZone = TimeZone(secondsFromGMT: 0)
+        createdAt = dateFormatter.string(from: entry.date ?? Date())
+        
+        utcOffset = nil
+        timestamp = Int((entry.date ?? Date()).timeIntervalSince1970)
+        enteredBy = "xdrip iOS"
```

Comment:
Don't copy-paste this value, define it as a constant (it is also used in other places).